enshrine-esdt-safe/src/lib.rs

```diff
@@ -24,9 +24,34 @@ pub trait EnshrineEsdtSafe:
     + common::storage::CommonStorage
 {
     #[init]
-    fn init(&self, is_sovereign_chain: bool) {
+    fn init(
+        &self,
+        is_sovereign_chain: bool,
+        wegld_identifier: Option<TokenIdentifier>,
+        sov_token_prefix: Option<ManagedBuffer>,
+    ) {
         self.is_sovereign_chain().set(is_sovereign_chain);
         self.set_paused(true);
+
+        if !is_sovereign_chain {
+            match wegld_identifier {
+                Some(identifier) => {
+                    require!(
+                        identifier.is_valid_esdt_identifier(),
+                        "Sent Identifier is not valid"
+                    );
+
+                    self.wegld_identifier().set(identifier);
+                }
+
+                None => sc_panic!("WEGLG identifier must be set in Mainchain"),
+            }
+
+            match sov_token_prefix {
+                Some(prefix) => self.sovereign_tokens_prefix().set(prefix),
+                None => sc_panic!("Sovereign Token Prefix must be set in Mainchain"),
+            }
+        }
```

Comment:
Can revert condition here to do some de-nesting, i.e.
```
if is_sovereign_chain {
    return;
}

// code that was inside if
```

Also, since you need "Some" for both Option, why not take "TokenIdentifier" and "ManagedBuffer" directly in the init call?

Reply:
i agree with the de-nesting but I don't quite understand what you mean by taking those directly in the init call